Monika-After-Story/MonikaModDev
# Better pen name usage

closes #4216

This allows the good, bad, and awkward nickname lists to be used by any topic for the purpose of Monika being able to properly react in `monika_penname` and `monika_name`. In `monika_penname`, while you can still pick a penname that is in one of the negative lists, she will no longer refer to nicknames that include these words as 'lovely.' For `monika_name` instead of comparing `mcname` with `player` (which doesn't really work that well since `mcname` is changed every time `player` is changed, plus people that change their name may not want their old name brought back up again) she will now compare it with your pen name if you have one.

## Testing

- verify the good/bad/awkward nickname lists still work as intended when changing the player name and Monika's nickname.

- verify `monika_penname` now properly reacts to the different levels of names.

- verify `monika_name` reacts properly to the different levels of names.
